backend/app/util.py

```diff
@@ -26,10 +29,18 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-# FIXME: Idk how to return proper HTTPException's to the frontend
 def decode_jwt(encoded_jwt: str):
-    idinfo = id_token.verify_oauth2_token(
-        encoded_jwt, requests.Request(), get_settings().google_client_id
-    )
-
-    return GoogleAuth(**idinfo)
+    try:
+        idinfo = id_token.verify_oauth2_token(
+            encoded_jwt, requests.Request(), get_settings().google_client_id
+        )
+        return GoogleAuth(**idinfo)
```

Comment:
Maybe we should have Pydantic's `ValidationError` as well

`from pydantic import ValidationError`
```suggestion
        return GoogleAuth(**idinfo)
    
    except ValidationError as e:
        raise HTTPException(status_code=498, detail=f"Could not parse Google's response: {e}")
```

Reply:
Pretty sure FastAPI does this by default with the endpoints `response_model`.

Reply:
But if im wrong lets add it in the future, im just merging for now